src/GuzzleHttp/Middleware.php

```diff
@@ -35,7 +35,7 @@ public static function cache(StorageAdapterInterface $adapter, $debug = false)
         return new CacheMiddleware($adapter, $debug);
     }
 
-    public static function history(\SplObjectStorage $container)
+    public static function history(\ArrayObject $container)
```

Comment:
That's a lot of reference. The code wouldn't be simpler but more complex to understand